### ISSUE TITLE
Editor freeze (MSI Afterburner overlay)

### ISSUE THREAD

**Issue by SmartySmart702** [godotengine/godot#62215]
### Godot version

4.0 alpha 10

### System information

Win 10, GTX 1070

### Issue description

If MSI Afterburner + Riva Tuner is active to view some stats the editor freezes pretty quickly after doing something.
I noticed that if I e.g. click on "Help" and then move my mouse down (where now the links should be listed) I can click and open them even if they are not visible so its not a 100% freeze (maybe refreshing related idk). 
Note that MSI Afterburner needs to be started before starting Godot. If I start MSI Afterburner when my project is already open the issue occurs less predictable (sometimes immediately, sometimes after a while).

Problem is gone if "Single Window Mode" is enabled.

Not sure if this is related or another issue but sometimes I cannot change collision layers/masks. There are more things like this where I cannot select, enable or change something but I dont remember what exactly.

Maybe related to #60073

### Steps to reproduce

Run a minimal project.

**Comment by Chaosus** [comment]
Maybe we should add some process checking before engine running to suggest user to end such processes (if this is unfixable on the engine side).

**Comment by IPcorps** [comment]
Same problem, plus BSOD (version 3.4.4): https://github.com/godotengine/godot/issues/18871#issuecomment-1158536456

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/57937 (same cause).

**Comment by AlexUnwinder** [comment]
> Maybe we should add some process checking before engine running to suggest user to end such processes (if this is unfixable on the engine side).

I could add the application to built-in RTSS hook exceptions list, but it is a bit troublesome in case of Godot becasue you use different executable name for each build and RTSS profiles are exact filename based. Alternately, you may add the following to your executable to prevent RTSS from hooking it:

extern "C"
{
	__declspec(dllexport) DWORD RTSSHooksCompatibility = 0x00000000;
}

**Comment by IPcorps** [comment]
I think this is a problem with Afterburner, not Godot, and they should fix it. In my case, I just started using HWiNFO + Riva Tuner and everything works fine (no BSODs or freezes at all).

**Comment by AlexUnwinder** [comment]
> I think this is a problem with Afterburner, not Godot, and they should fix it. In my case, I just started using HWiNFO + Riva Tuner and everything works fine (no BSODs or freezes at all).

Wrong. Afterburner is not a problem source in this case. It doesn't hook external processes (including Godot) in any form. It is overclocking application and improper hardware management (including too agressive undervolting, overclocking or altering default fan speed curve) is BSOD reason. I won't be able to "fix" it.
RivaTuner's overlay incompatibility with Godot is a key thing in context of this thread. I don't target my overlay implementation on applications using with multiple Vulkan swapchains, so does the majority (if not all) other Vulkan overlay implementations. So excluding it from RTSS hooking is the only solution.
